### PR TITLE
feat: added quizzes page, reducer, context

### PR DESCRIPTION
**Pull request by poojakelkar**
chrome-capture-2022-3-7

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -6,19 +7,24 @@ import Questions from "./pages/Questions";
 import Quizzes from "./pages/Quizzes";
 import Result from "./pages/Result";
 import Signup from "./pages/Signup";
```
Comment (xii-vi) [inline src/App.jsx:10]: _You could export these into an index file. _

File: src/App.jsx [hunk 2]
```diff
@@ -6,19 +7,24 @@ import Questions from "./pages/Questions";
...
+import { QuizContext } from "./store/context";
+import { initialState, reducer } from "./store/reducer";
 
 function App() {
+  const [state, dispatch] = useReducer(reducer, initialState);
```
Comment (xii-vi) [inline src/App.jsx:15]: try to give a generic name to the state dispatch and reducer function for better readability.

File: src/store/reducer.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import { types } from "./constants";
+import data from "./data.json";
+
+export const reducer = (currentState, action) => {
+  console.log({ ...currentState }, "previous");
```
Comment (xii-vi) [inline src/store/reducer.js:5]: remove console log after usage

**Review by xii-vi — APPROVED** [review]
_Code is very clean and oragnised.
Left a few sanity comments.
Really like the use of Nullish coalescing operator._

**Review by HrishikeshMahalle — APPROVED** [review]
The code Looks good.

File: src/App.jsx (hunk 1 above)
Comment (poojakelkar) [reply src/App.jsx:10]: yeah will do it after submission thanks for PR

File: src/App.jsx (hunk 2 above)
Comment (poojakelkar) [reply src/App.jsx:15]: Okk

**Pull request merged**